source/val/function.h

```diff
@@ -174,6 +174,9 @@ class Function {
   /// Returns the block predecessors function for the augmented CFG.
   GetBlocksFunction AugmentedCFGPredecessorsFunction() const;
 
+  /// Returns the control flow nesting depth of the given basic block.
+  int GetBlockDepth(BasicBlock* bb);
```

Comment:
Hmm.  Maybe the algorithm only works if you know you have structured control flow.  So I suggest you guard the check by whether you have Shader capability.